modules/player/src/main/java/unnamed/mmo/DamageTagList.java

```diff
@@ -0,0 +1,31 @@
+package unnamed.mmo;
+
+import net.minestom.server.item.ItemStack;
+import net.minestom.server.tag.Tag;
+
+public class DamageTagList {
+
+    public static final Tag<Double> ENTITY_WEIGHT_TAG = Tag.Double("mob-weight");
+    public static final Tag<Integer> WEAPON_TYPE_TAG = Tag.Integer("weapon-weight");
+
+    public static WeaponWeight getWeaponWeight(ItemStack itemStack) {
```

Comment:
These should be part of the weapon component, then you can get them from an item stack something like

```java
Item item = Item.fromItemStack(itemStack);
Weapon weapon = item.getComponent(Weapon.class);
weapon.weight() // or something
```

Reply:
There are two justifications here:
1. When an item is serialized eventually it wont bring along all of its nbt data (need to work this into component handlers)
2. If its on the Item definition then its just stored once for all instances of the item, rather than per item. We will perhaps need to look into modifiers for this value (eg from an enchantment), but that system is super non final.